IanGRap/Squish
# Block Spawner

Contributors: **James Petnic**, **Alfredo Cuevas**

High Level Goals: 
Create a block spawner that takes shapes and spawns blocks based on the players' positions.

Important Files:
Spawner.cs: Calls a coroutine that spawn's a block over a player at a given interval.
PlayerTracker.cs: Finds all the players and returns the Transform of one of the players
[BlockSpawner] prefab: Contains scripts for spawning blocks. The position affects how high blocks spawn. 
Prefabs Created: FiveBlockShape, ThreeBlockRow, X1.5, X2

Adjustable values: 
**Spawn Timer**: The time interval between blocks spawn calls

**Size Of Arena**: Size of the play stage. This value is used to determine when blocks need to be shifted.

**Block Size**: This should be the size of the blocks used in shapes. It determines how much shapes are shifted when they are out of range.

**Shape Prefabs Array**: This should contain all the shapes that you want to spawn. The shapes should all use blocks that are the same size. ex) 1x1, 1.5x1.5, 2x2 

**Spawn On Random Player**: If selected the blocks will spawn on player positions but will not be in any order. 

Expected Behavior: 
* If players aren't on the edges of the arena the blocks should spawn almost directly above them. 
* If the players are on the edges then the blocks will be shifted over by 1 block so that they remain inside of the play field

How to Test: 
Run the Block Spawning Scene. (All the players are bound to Player 1). Move around. The blocks should spawn above the players in a consistent order. If you move a player to the edge of the map blocks should spawn shifted over. 
Then try setting the SpawnOnRandomPlayer variable and see if the blocks spawn above players in a random order.
You can add or delete objects from the ShapePrefabs Array to change the shapes that spawn. Don't add prefabs with blocks of different scales. Try adding in a third prefab - NewShape. The new shape should be added to the rotation.
